validators/cgroup_validator_linux.go

```diff
@@ -44,10 +47,96 @@ func (c *CgroupsValidator) Name() string {
 }
 
 const (
-	cgroupsConfigPrefix = "CGROUPS_"
-	unifiedMountpoint   = "/sys/fs/cgroup"
+	cgroupsConfigPrefix      = "CGROUPS_"
+	defaultUnifiedMountpoint = "/sys/fs/cgroup"
+	mountsFilePath           = "/proc/mounts"
+	mountInfoPath            = "/proc/self/mountinfo"
 )
 
+// 1. check /proc/self/mountinfo if the default mount point is available
+// 2. check /proc/mounts and return the cgroup dir
+func getUnifiedMountpoint() string {
```

Comment:
make this function return a string, error instead of calling printf() in it.

Reply:
Do we need this warning? 
https://github.com/kubernetes/system-validators/pull/41/files#diff-0cb89d807d0cf7f118534018122b93367efce21da4e1650705f6c991872ae3e4R63

Reply:
i think the url you posted is not correct ^
but basically if this func returns err the caller can decide if it's a warning or error.

Reply:
https://github.com/kubernetes/system-validators/pull/41/files#r1800594545 This link.

Reply:
print should only be done from the main Validate() function.
let's try to show this warning on the top caller.

once you update i will check again.

Reply:
return it as a warning.

Updated.